src/interpreter_funcs/voxel_interpolated_union.rs

```diff
@@ -0,0 +1,260 @@
+                separate mesh shells: one for outer surface, the other for inner surface of \
+                hollow watertight mesh.",
+                refinement: ParamRefinement::Boolean(BooleanParamRefinement {
+                    default_value: true,
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Factor",
+                description: "Proportional interpolation factor between the first and the second mesh.\n\
+                \n\
+                Factor = 0.0: the result is equal to the first mesh.\n\
+                Factor = 0.5: the result is half way between the first and the second mesh.\n\
+                Factor = 1.0: the result is equal to the second mesh.",
+                refinement: ParamRefinement::Float(FloatParamRefinement {
+                    default_value: Some(0.0),
+                    min_value: Some(0.0),
+                    max_value: Some(1.0),
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Prevent Unsafe Settings",
+                description: "Stop computation and throw error if the calculation may be too slow.",
+                refinement: ParamRefinement::Boolean(BooleanParamRefinement {
+                    default_value: true,
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Bounding Box Analysis",
+                description: "Reports basic and quick analytic information on the created mesh.",
+                refinement: ParamRefinement::Boolean(BooleanParamRefinement {
+                    default_value: true,
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Detailed Mesh Analysis",
+                description: "Reports detailed analytic information on the created mesh.\n\
+                              The analysis may be slow, therefore it is by default off.",
+                refinement: ParamRefinement::Boolean(BooleanParamRefinement {
+                    default_value: false,
+                }),
+                optional: false,
+            },
+        ]
+    }
+
+    fn return_ty(&self) -> Ty {
+        Ty::Mesh
+    }
+
+    fn call(
+        &mut self,
+        args: &[Value],
+        log: &mut dyn FnMut(LogMessage),
+    ) -> Result<Value, FuncError> {
+        let mesh1 = args[0].unwrap_mesh();
+        let mesh2 = args[1].unwrap_mesh();
+        let voxel_dimensions = Vector3::from(args[2].unwrap_float3());
+        let fill = args[3].unwrap_boolean();
+        let factor = args[4].unwrap_float();
+        let error_if_large = args[5].unwrap_boolean();
+        let analyze_bbox = args[6].unwrap_boolean();
+        let analyze_mesh = args[7].unwrap_boolean();
+
+        if voxel_dimensions
+            .iter()
+            .any(|dimension| approx::relative_eq!(*dimension, 0.0))
```

Comment:
Let's check for negative values here.

Dejavu again. Would be good if you checked all your future PRs for this, so we don't have to do this every time.